### PR TITLE
Social: Add instagram aspect-ratio validation for images

### PR DESCRIPTION
**Pull request by gmjuhasz**
At the moment we only check aspect-ratio for videos. This should be fixed, this change adds the needed logic to correctly check the width and aspect-ratio requirements for Instagram

## Proposed changes:

*

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

No.
## Testing instructions:

* On trunk turn off auto-conversion on your site.
* Find an image that [does not meet the requirements](https://developers.facebook.com/docs/instagram-api/reference/ig-user/media/#image-specifications) for the aspect-ratio validation of Instagram
* Uploading that as attached media - if the filesize is correct - should not show errors.
* Check out this PR. Applying the same image should result in a warning.

### CODE REVIEW

File: projects/js-packages/publicize-components/src/hooks/use-media-restrictions/restrictions.js [hunk 1]
```diff
@@ -126,6 +132,12 @@ export const RESTRICTIONS = {
 		allowedMediaTypes: [ 'image/jpg', 'image/jpeg', MP4, MOV, VIDEOPRESS ],
 		image: {
 			maxSize: 8,
+			minWidth: 320,
+			maxWidth: 1440,
+			aspectRatio: {
+				min: 4 / 5,
+				max: 1.91 / 1,
```
Comment (pablinos) [inline projects/js-packages/publicize-components/src/hooks/use-media-restrictions/restrictions.js]: Do you think we should just keep this as 1.91?

```suggestion
				max: 1.91,
```

File: projects/js-packages/publicize-components/src/hooks/use-media-restrictions/index.js [hunk 2]
```diff
@@ -138,7 +158,7 @@ const getValidationError = ( metaData, mediaData, serviceName, shouldUploadAttac
 				mediaData?.height,
 				restrictions.video
 		  )
-		: getImageValidationError( sizeInMb, restrictions.image.maxSize );
+		: getImageValidationError( sizeInMb, mediaData?.width, mediaData?.height, restrictions.image );
```
Comment (pablinos) [inline projects/js-packages/publicize-components/src/hooks/use-media-restrictions/index.js]: Can `mediaData` be null/undefined? If so, then we're not guarding against it in the `getImageValidationError` function and might cause problems with the ratio calculation.

image

Seems like it depends on what is null. Probably it's ok, but getting back NaN I could see might cause problems.

File: projects/js-packages/publicize-components/src/components/media-picker/index.js [hunk 3]
```diff
@@ -75,15 +80,30 @@ export default function MediaPicker( {
...
 							></VideoPreview>
 						) : (
-							<ResponsiveWrapper naturalWidth={ width } naturalHeight={ height } isInline>
-								<img src={ sourceUrl } alt="" className={ styles[ 'preview-image' ] } />
+							<ResponsiveWrapper
+								naturalWidth={ previewWidth || width }
+								naturalHeight={ previewHeight || height }
+								isInline
+							>
+								<img src={ previewUrl } alt="" className={ styles[ 'preview-image' ] } />
```
Comment (pablinos) [inline projects/js-packages/publicize-components/src/components/media-picker/index.js]: Should we also use `sourceUrl` here if we're also falling back to `width` and `height`?

```suggestion
								<img src={ previewUrl || sourceUrl } alt="" className={ styles[ 'preview-image' ] } />
```

File: projects/js-packages/publicize-components/src/components/media-picker/index.js [hunk 4]
```diff
@@ -27,6 +27,7 @@ const clickHandler = open => e => {
  * @param {object} props.mediaDetails - The details of the media for preview
  * @param {Function} props.onChange - A callback that can be passed to parent for validation
  * @param {string} props.wrapperClassName - A class name to be added to the wrapper
+ * @param {object} props.allowedMediaTypes - Custom allowed media types
```
Comment (pablinos) [inline projects/js-packages/publicize-components/src/components/media-picker/index.js:30]: Are we using this? Should this be passed in when auto conversion is not enabled?

**Review by pablinos — COMMENTED** [review]
This is pretty close I think. It's working well for me, but maybe a couple of tweaks

File: projects/js-packages/publicize-components/src/components/media-picker/index.js (hunk 4 above)
Comment (gmjuhasz) [reply projects/js-packages/publicize-components/src/components/media-picker/index.js:30]: It's used here: fbhepr%2Skers%2Swrgcnpx%2Scebwrpgf%2Swf%2Qcnpxntrf%2Schoyvpvmr%2Qpbzcbaragf%2Sfep%2Spbzcbaragf%2Sfbpvny%2Qvzntr%2Qtrarengbe%2Scnary%2Szbqny.wf%3Se%3Qro7374p5%26zb%3Q898%26sv%3Q22%2382-og

It was not working before, this fixes it

**Review by manzoorwanijk — DISMISSED** [review]
This looks good to me.

**Review by pablinos — APPROVED** [review]

**Pull request merged**